xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp

```diff
@@ -6,178 +6,187 @@
-    m_overlay[1]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[1]));
+    m_adapter->AddSubtitle(m_strDebug[1].c_str(), 0., 5000000.);
   }
   if (video.metaLight != m_strDebug[2])
   {
     m_strDebug[2] = video.metaLight;
-    if (m_overlay[2])
-      m_overlay[2]->Release();
-    m_overlay[2] = new CDVDOverlayText();
-    m_overlay[2]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[2]));
+    m_adapter->AddSubtitle(m_strDebug[2].c_str(), 0., 5000000.);
   }
   if (video.shader != m_strDebug[3])
   {
     m_strDebug[3] = video.shader;
-    if (m_overlay[3])
-      m_overlay[3]->Release();
-    m_overlay[3] = new CDVDOverlayText();
-    m_overlay[3]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[3]));
+    m_adapter->AddSubtitle(m_strDebug[3].c_str(), 0., 5000000.);
   }
   if (render.renderFlags != m_strDebug[4])
   {
     m_strDebug[4] = render.renderFlags;
-    if (m_overlay[4])
-      m_overlay[4]->Release();
-    m_overlay[4] = new CDVDOverlayText();
-    m_overlay[4]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[4]));
+    m_adapter->AddSubtitle(m_strDebug[4].c_str(), 0., 5000000.);
   }
   if (render.videoOutput != m_strDebug[5])
   {
     m_strDebug[5] = render.videoOutput;
-    if (m_overlay[5])
-      m_overlay[5]->Release();
-    m_overlay[5] = new CDVDOverlayText();
-    m_overlay[5]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[5]));
+    m_adapter->AddSubtitle(m_strDebug[5].c_str(), 0., 5000000.);
   }
 
-  for (int i = 0; i < 6; i++)
-    m_overlayRenderer.AddOverlay(m_overlay[i], 0, 0);
+  auto overlay = m_adapter->CreateOverlay();
+  m_overlayRenderer.AddOverlay(overlay, 1000000., 0);
 }
 
 void CDebugRenderer::Render(CRect& src, CRect& dst, CRect& view)
 {
+  if (!m_isConfigured)
+    return;
+
   m_overlayRenderer.SetVideoRect(src, dst, view);
   m_overlayRenderer.Render(0);
 }
 
 void CDebugRenderer::Flush()
 {
+  if (!m_isConfigured)
+    return;
+
   m_overlayRenderer.Flush();
 }
 
 CDebugRenderer::CRenderer::CRenderer() : OVERLAY::CRenderer()
 {
-  m_font = "__debug__";
-  m_fontBorder = "__debugborder__";
 }
 
 void CDebugRenderer::CRenderer::Render(int idx)
 {
-  std::vector<COverlay*> render;
   std::vector<SElement>& list = m_buffers[idx];
-  float posY = 0.0f;
   for (std::vector<SElement>::iterator it = list.begin(); it != list.end(); ++it)
   {
     COverlay* o = nullptr;
 
     if (it->overlay_dvd)
-      o = Convert(it->overlay_dvd, it->pts);
-
-    if (!o)
-      continue;
+    {
+      CDVDOverlayLibass* ovAss = static_cast<CDVDOverlayLibass*>(it->overlay_dvd);
+      if (!ovAss || !ovAss->GetLibassHandler())
+        continue;
 
-    COverlayText *text = dynamic_cast<COverlayText*>(o);
-    if (text)
-      text->PrepareRender("arial.ttf", 1, 100, 15, 0, m_font, m_fontBorder, UTILS::COLOR::NONE,
-                          m_rv);
+      KODI::SUBTITLES::subtitlesStyle subStyle;
+      bool updateStyle = !ovAss->GetLibassHandler()->IsStyleInitialized();
```

Comment:
I am not a big fan of this. Whenever you need to ask a component about its internal state it's usually a sign the design logic can be improved. Can you figure out a way of writing this so that the managers (overlayrenderer and debugrenderer) are responsible for tracking the state instead? Probably you can initialize the style right on the adapter->initialize() and store it on the manager as a member. Then, on settingschanged observed callback you update the member style in the manager and flag the convert() method the style needs to be updated on the handler side

Reply:
the adapter here has nothing to do with it,
but i will try to seeto remove IsStyleInitialized and manage it directly internally libass class
so we will no longer have this to do here (and also in overlayrenderer)

Reply:
I remember now,
no this need to be handled as is, to avoid an CPU overload in creating styles unnecessarily at every frame

Reply:
because you create the style inside the debugrenderer/overlayrenderer class

Reply:
ah you mean to move this variable in the debugrenderer/overlayrenderer class?

Reply:
I mean the manager should not ask the state of a component it is in charge of. I am pretty sure we can figure out a way of refactoring this without asking the state and still have the same performance levels

Reply:
I have to try see, because when i have done this, i encountered many problems when you switch subtitles with another subtitle track _while in playback_, but i don't remember exactly now

Reply:
Tried check again, on debugrenderer we can think to remove it,
but in the overlayrenderer is needed, or will cause a big mess when you choose another subtitles tracks (eg. external file etc..) while playing,
because when you are in playback, the overlayrenderer in always the same class instance, the overlay's (that now is one single, replied to each frame) are async and you have no chance to know when you are switching to another libass instance that require the style to be set

Reply:
so without knowing the status of the libass component, IMO you can't solve it because is a libass internal state,
if you know another possible way, try push in your github the changes, I don't see any other solution

Reply:
I'll try to take a look in ~ 1 week when I am back near a computer

Reply:
your provided example works well thank you
this is also done